public/externalLibs/sound/sounds.js

```diff
@@ -576,15 +549,45 @@ function sawtooth_sound(freq, duration) {
 
 /**
  * plays a given sound without regard if a sound is already playing
- * @param {sound} sound - given sound
- * @returns {undefined} undefined
+ * @param {Sound} sound - given sound
+ * @returns {undefined}  undefined
  */
 function play_concurrently(sound) {
-    // Discretize the input sound
-    var data = discretize(get_wave(sound), get_duration(sound));
-    _safeaudio = raw_to_audio(data);
+    // Type-check sound
+    if (!is_sound(sound)) {
+        throw new Error(`play_concurrently is expecting sound, but encountered ${sound}`);
+    }  else if (get_duration(sound) <= 0) {
+        return;
+    } else {
+        // Instantiate audio context if it has not been instantiated
+        if (!_audioplayer) {
+            init_audioCtx();
+        }
+
+        // Create mono buffer
+        let theBuffer = _audioplayer.createBuffer(1, FS * get_duration(sound), FS);
+        let channel = theBuffer.getChannelData(0);
+
+        // Discretize the function and clip amplitude
+        let temp;
+        const wave = get_wave(sound);
+        for (let i = 0; i < channel.length; i++) {
+            temp = wave(i/FS);
+            channel[i] = temp > 1 ? 1 : temp < -1 ? -1 : temp;
```

Comment:
I see that all of the `raw_to_audio` functionality except the clipping portion are removed
`channel[i] = temp > 1 ? 1 : temp < -1 ? -1 : temp;`
The quantization and exponential filter aspects from `raw_to_audio` are removed, is this change intended?

Reply:
Quantize was removed as webaudio buffers are stored as float32 arrays.
Removing the exponential decay fade-out was unintended, i'll re-add it back in